src/global/Pattern.h

```diff
@@ -20,10 +21,10 @@
 #include "util/File.h"
 #include "util/Generator.h"
 #include "util/Iterators.h"
+#include "util/ResetWhenMoved.h"
 #include "util/Serializer/FileSerializer.h"
 #include "util/Serializer/SerializeVector.h"
 #include "util/TypeTraits.h"
-#include "util/UninitializedAllocator.h"
 
 typedef uint32_t PatternID;
```

Comment:
unrelated, but we can use `using` instead for more idiomatic C++. (And maybe it doesn't need to be in the global namespace if it's only being used for `NO_PATTERN` below, not sure)